src/org/mozilla/javascript/AbstractEcmaObjectOperations.java

```diff
@@ -0,0 +1,111 @@
+package org.mozilla.javascript;
+
+/**
+ * Abstract Object Operations as defined by EcmaScript
+ *
+ * @see <a href="https://262.ecma-international.org/11.0/#sec-operations-on-objects">Abstract
+ *     Operations - Operations on Objects</a>
+ */
+class AbstractEcmaObjectOperations {
+    static enum INTEGRITY_LEVEL {
+        FROZEN,
+        SEALED
+    }
+
+    /**
+     * Implementation of Abstract Object operation testIntegrityLevel as defined by EcmaScript
+     *
+     * @param o
+     * @param level
+     * @return boolean
+     * @see <a
+     *     href="https://262.ecma-international.org/11.0/#sec-testintegritylevel">TestIntegrityLevel</a>
+     */
+    static boolean testIntegrityLevel(Object o, INTEGRITY_LEVEL level) {
+        ScriptableObject obj = ScriptableObject.ensureScriptableObject(o);
+        Context cx = Context.getCurrentContext();
```

Comment:
It looks like this function is called from various "execIdCall" methods, which means that the Context should already be in the parameter list. I'd prefer if this method took Context as the first parameter too. That saves us from figuring out what to do in the unlikely event that the Context is not properly on the thread (which is incorrect but could still happen) and it also saves us from having to look up the context from the thread slot where it's stored, which is I assume a more expensive operation than having it passed as a parameter.